include/libcyphal/transport/can/transport.hpp

```diff
@@ -0,0 +1,219 @@
+/// @copyright
+/// Copyright (C) OpenCyphal Development Team  <opencyphal.org>
+/// Copyright Amazon.com Inc. or its affiliates.
+/// SPDX-License-Identifier: MIT
+
+#ifndef LIBCYPHAL_TRANSPORT_CAN_TRANSPORT_HPP_INCLUDED
+#define LIBCYPHAL_TRANSPORT_CAN_TRANSPORT_HPP_INCLUDED
+
+#include "media.hpp"
+#include "libcyphal/transport/transport.hpp"
+#include "libcyphal/transport/multiplexer.hpp"
+
+#include <canard.h>
+
+#include <algorithm>
+
+namespace libcyphal
+{
+namespace transport
+{
+namespace can
+{
+
+class ICanTransport : public ITransport
+{};
+
+namespace detail
+{
+class TransportImpl final : public ICanTransport
+{
+public:
+    TransportImpl(cetl::pmr::memory_resource&            memory,
+                  libcyphal::detail::VarArray<IMedia*>&& media_array,
```

Comment:
why are we aliasing variable_length_array but not span? What's the logic we're using to make a type a libcyphal type rather than an external one?

Reply:
>why are we aliasing variable_length_array but not span?

It's a workaround for https://github.com/OpenCyphal/CETL/issues/115 due to be undone.

@serges147 can we please use span as originally planned, considering that CETL has it fixed now?

Reply:
Please let me fix this in separate dedicated PR - I have already 3 work-in-progress private branches. Changing this right now will make me conflicts to resolve in all these WIP branches. As a sure thing promise here is explicit issue: https://github.com/OpenCyphal-Garage/libcyphal/issues/349

Reply:
ok